blockchain/src/chain_store.rs

```diff
@@ -161,7 +287,18 @@ impl ChainStore {
 
         // Store body if requested.
         if include_body {
-            txn.put_reserve(&self.block_table, hash, &chain_info.head);
+            // If block is on main chain we store it in the `pushed` block table: we should be able to retrieve
+            // the block from the `chain_info` table and the history store.
+            // Otherwise it goes completely to the `stored` block table.
+            if chain_info.on_main_chain {
+                txn.put_reserve(
+                    &self.pushed_block_table,
+                    hash,
+                    &PushedBlock::from(chain_info.head.clone()),
```

Comment:
The `clone` here is unnecessarily expensive since it clones the full block, but `PushedBlock::from` only needs the `equivocation_proofs` for micro blocks. It would be more efficient to implement `From<&Block>` and clone only the `equivocation_proofs` within the trait implementation.